### PR TITLE
Added Presence for AniMixPlay.to

### PR DESCRIPTION
**Pull request by jackkelly-bayliss**
Requested By https://github.com/PreMiD/Presences/issues/2600
bYlWt6Uo

### CODE REVIEW

File: websites/A/AniMixPlay/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "author": {
+    "name": "Jack Hansen",
+    "id": "140590672657776641"
+  },
+  "service": "AniMixPlay",
+  "description": {
+    "en": "AniMixPlay is a website that allows you to watch Anime from all around the world, completely free."
+  },
+  "url": "animixplay.to",
+  "version": "1.0.1",
```
Comment (ririxidev) [inline websites/A/AniMixPlay/dist/metadata.json:11]: 1.0.0

File: websites/A/AniMixPlay/presence.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          paused: any,
+          timestamps: any,
+          video: HTMLVideoElement;
+        
+        const iframe = document.getElementById("iframeplayer");
+        if (iframe) {
+          video = iframe.contentWindow.document.getElementsByTagName("video")[0];
+        } else {
+          video = document.querySelector("#playercontainer > div > div.plyr__video-wrapper > video");
+        }
```
Comment (ririxidev) [inline websites/A/AniMixPlay/presence.ts:72]: if you get iframe, use iframe.ts thanks :)

File: websites/A/AniMixPlay/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "logo": "https://i.imgur.com/eMXyO0Y.png",
```
Comment (ririxidev) [inline websites/A/AniMixPlay/dist/metadata.json:12]: maybe more quality logo since this one has white bar at left

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/A/AniMixPlay/presence.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused",
+    browsing: "presence.activity.browsing"
+});
+
+/**
+ * Get Timestamps (Credit to Bas950)
+ * @param {Number} videoTime Current video time seconds
+ * @param {Number} videoDuration Video duration seconds
+ */
```
Comment (Bas950) [inline websites/A/AniMixPlay/presence.ts:15]: no need to add this either.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AniMixPlay/dist/metadata.json (hunk 3 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/dist/metadata.json:12]: I can't see a white bar on this logo, could you send a screenshot please? :0

Comment (Bas950) [reply websites/A/AniMixPlay/dist/metadata.json:12]: image
Looking good to me :/

File: websites/A/AniMixPlay/presence.ts (hunk 4 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/presence.ts]: Removed

File: websites/A/AniMixPlay/dist/metadata.json (hunk 1 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/dist/metadata.json]: Changed

File: websites/A/AniMixPlay/dist/metadata.json (hunk 3 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/dist/metadata.json:12]: #BlameGoogleDown
b4QzjyXF

Comment (ririxidev) [reply websites/A/AniMixPlay/dist/metadata.json:12]: Screen Shot 2020-12-14 at 15 34 53
for me it looks like this

File: websites/A/AniMixPlay/presence.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            presenceData.state = "" + urlParams.get("season").toLowerCase() + " " + urlParams.get("year").toLowerCase() + " anime";
+            presenceData.smallImageKey = "browsing-v1";
+            presenceData.smallImageText = "Exploring...";
+        } else {
+            presenceData.startTimestamp = browsingTimer;
+            presenceData.details = "Currently browsing...";
+            presenceData.smallImageKey = "browsing-v1";
+            presenceData.smallImageText = (await strings).browsing;
+        }
+      } else if (document.location.pathname.includes("/v4/") || document.location.pathname.includes("/v1/") || document.location.pathname.includes("/v7/")) {
```
Comment (sakamotodesu1) [inline websites/A/AniMixPlay/presence.ts:55]: better use regex to match all stream pages url ( /v1/ , /v2/ , /v3/ , ... , /v9/ )
`/^\/v.\//.test(document.location.pathname)`
or just add match for url with other number.

**Review by sakamotodesu1 — CHANGES_REQUESTED** [review]

File: websites/A/AniMixPlay/presence.ts (hunk 5 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/presence.ts]: Added

File: websites/A/AniMixPlay/presence.ts (hunk 2 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/presence.ts]: Migrated :)

File: websites/A/AniMixPlay/presence.ts [hunk 6]
```diff
@@ -53,24 +79,19 @@ presence.on("UpdateData", async () => {
...
-          duration: any,
-          paused: any,
-          timestamps: any,
-          video: HTMLVideoElement;
+        let video: HTMLVideoElement;
         
         const iframe = document.getElementById("iframeplayer");
-        if (iframe) {
-          video = iframe.contentWindow.document.getElementsByTagName("video")[0];
+        if (iFrameVideo === true) {
```
Comment (ririxidev) [inline websites/A/AniMixPlay/presence.ts:85]: big js protip:
```javascript
if(iFrameVideo) {
```
is enough

**Review by ririxidev — CHANGES_REQUESTED** [review]
fix codacy errors

File: websites/A/AniMixPlay/presence.ts (hunk 6 above)
Comment (jackkelly-bayliss) [reply websites/A/AniMixPlay/presence.ts]: Resolved

**Review by ririxidev — CHANGES_REQUESTED** [review]
upload new proof of working presence with these changes

**Comment by jackkelly-bayliss** [comment]
Updated proof of presence:
Neeeo7qK

**Review by ririxidev — CHANGES_REQUESTED** [review]
also, codacy is showing errors right. you can't access variables when you do them in ifs etc.
image

**Review by ririxidev — CHANGES_REQUESTED** [review]
send last proof of working presence and ready to accept! :D

**Comment by jackkelly-bayliss** [comment]
> send last proof of working presence and ready to accept! :D

Had to add back a '/' in the regex expression, if Codacy reports anything - I'm shooting my fox.

LwBFan4q

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**